src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -69,8 +70,35 @@ public ModelStateDictionary ModelState
             }
         }
 
+        /// <summary>
+        /// Gets or Sets the <see cref="Mvc.ActionContext"/> object for execution of action which has been selected
+        /// as part of an HTTP request.
+        /// </summary>
+        /// <remarks>
+        /// By default, this property is activated when <see cref="IControllerActivator"/> activates controllers.
+        /// However, when controllers are directly instantiated in user codes, this property is initialized with
+        /// an empty <see cref="Mvc.ActionContext"/>.
+        /// </remarks>
         [Activate]
-        public ActionContext ActionContext { get; set; }
+        public ActionContext ActionContext
+        {
+            get
+            {
+                // This should run only for the controller unit test scenarios
+                _actionContext = _actionContext ?? new ActionContext();
+                return _actionContext;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw
+                        new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(ActionContext));
```

Comment:
Put the breaks inside the parameter list. 

```
throw new ArgumentException(
    Resources.ArgumentCannotBeNullOrEmpty,
    nameof(ActionContext));
```

This is also not the correct error. You're not checking null or empty (that's for strings). Just use `ArgumentNullException`. Also please read the documentation, `nameof(ActionContext)` is not correct to the argument name.